pypact/input/inputdata.py

```diff
@@ -425,6 +425,32 @@ def _deserialize(self, f):
             The deserialization method
             f: file object
         """
-        pass
-
-
+        self.reset()
+
+        lines = f.readlines()
+        in_mass_section = False
+
+        for line in lines:
+            line = line.strip()
+            if line.startswith("MASS"):
+                # Parse the MASS line
+                parts = line.split()
+                if len(parts) < 3:
+                    raise PypactInvalidOptionException("Invalid MASS line format.")
+                self._inventoryismass = True
+                self._inventorymass.totalMass = float(parts[1])
+                num_elements = int(parts[2])
+                self._inventorymass.entries = []
```

Comment:
Again, you could use the [`addElement`](https://github.com/fispact/pypact/blob/b29640f6aebd5ad83e19c58997c284eb48e70aa5/pypact/input/inputdata.py#L262) method here instead. This way it checks that it is less than 100%.

Reply:
Sorry this should point to line 451.